File: Hard/10. Regular Expression Matching/1.js
/**
 * @param {string} s
 * @param {string} p
 * @return {boolean}
 */
var isMatch = function(s, p) {
    var sl=s.length,pl=p.length;
    var a=(i,j)=>i<sl?j<pl&&(j+1<pl&&p[j+1]==="*"?((s[i]===p[j]||p[j]===".")&&a(i+1,j))||a(i,j+2):(s[i]===p[j]||p[j]===".")&&a(i+1,j+1)):j+1<pl&&p[j+1]==="*"?a(i,j+2):j>=pl||(j+1<pl&&p[j+1]==="*");
    return a(0,0);
};

File: Easy/509. Fibonacci Number/1.rb
# @param {Integer} n
# @return {Integer}
@part1=(1+5**0.5)/2
@part2=(1-5**0.5)/2
@part3=5**-0.5
def fib(n)
    return (@part3*(@part1**n-@part2**n)).round
end
